http/middleware.go

```diff
@@ -52,7 +52,7 @@ func processRequest(tx types.Transaction, req *http.Request) (*types.Interruptio
 	if in != nil {
 		return in, nil
 	}
-	if req.Body != nil {
+	if req.Body != nil && req.Body != http.NoBody {
```

Comment:
Debugging I found that `req.Body == http.NoBody` happens frequently. Being [`a io.ReadCloser with no bytes`](https://pkg.go.dev/net/http) I think it can be excluded just like `nil`